micro-rdk/src/native/entry.rs

```diff
@@ -120,6 +135,117 @@ pub async fn serve_web_inner(
     srv.serve(robot).await;
 }
 
+#[cfg(feature = "provisioning")]
+async fn serve_provisioning_async<S>(
+    ip: Ipv4Addr,
+    exec: NativeExecutor,
+    info: ProvisioningInfo,
+    storage: S,
+    last_error: Option<String>,
+) -> Result<(AppClientConfig, NativeTlsServerConfig), Box<dyn std::error::Error>>
+where
+    S: CredentialStorage + Clone + 'static,
+    S::Error: Debug,
+    GrpcError: From<S::Error>,
+{
+    let hostname = format!(
+        "provisioning-{}-{}",
+        info.get_model(),
+        info.get_manufacturer()
+    );
+    let mut mdns = NativeMdns::new(hostname, ip)?;
+
+    let srv = ProvisioningServiceBuilder::new().with_provisioning_info(info);
+
+    let srv = if let Some(error) = last_error {
+        srv.with_last_error(error)
+    } else {
+        srv
+    };
+    let srv = srv.build(storage.clone());
+    let listen = TcpListener::bind(ip.to_string() + ":0")?;
```

Comment:
For native, does it make sense to only bind to one v4 ip if the device is multi-homed?

Reply:
My assumption is that the service will be discoverable via mDNS so if the server ip can be reached you should see it. Note that the assumption could be wrong since on your computer it didn't seem to works.